### PR TITLE
``@phenomic/core`` + ``@phenomic/api-client``: Make `path` parameter for query optional

### PR DESCRIPTION
**Pull request by MoOx**
Because we can.

### CODE REVIEW

**Comment by MoOx** [comment]
Note that I made this the quick and not really pretty way.

**Comment by MoOx** [comment]
Smaller diff https://github.com/phenomic/phenomic/pull/1173/files?w=1

File: packages/core/src/api/index.js [hunk 1]
```diff
@@ -142,6 +142,20 @@ function createServer(db: PhenomicDB, plugins: PhenomicPlugins) {
     }
   });
 
+  server.get("/item/*.json", async function(
+    req: express$Request,
+    res: express$Response
+  ) {
+    debug(req.url, JSON.stringify(req.params));
+    try {
+      const resource = await db.get(undefined, req.params["0"]);
```
Comment (bloodyowl) [inline packages/core/src/api/index.js]: maybe should we use `null` to express that the absence of value is wished?

File: packages/plugin-collector-files/src/index.js [hunk 2]
```diff
@@ -118,6 +118,10 @@ export default function() {
           debug(`collecting ${relativeKey} for path '${pathName}'`);
           return Promise.all([
             // full resource, not sorted
+            db.put(undefined, key, {
```
Comment (bloodyowl) [inline packages/plugin-collector-files/src/index.js]: same question about `null`

**Pull request merged**